### DISCUSSION TITLE
Need some sort of auto-purge of logged requests

### DISCUSSION THREAD

**Discussion by jishi** [craftcms/webhooks#57] [Ideas] [answered]
### Description

I have been running this plugin on Entry afterSave for two weeks, and it has already accumulated 67k rows which prevents me from backing up the database from the admin interface (and blocking access to the site). 

Preferably some retention setting (1 day, 1 week) and purge all entries that are done and older automatically. Also with correct indexes to avoid a cleanup to choke the whole database server :)

**Comment by mildlygeeky** [comment]
Seconding this - I was surprised to see how quickly the size of a Craft DB had grown, and assumed it must have been Neo or something. Was shocked to see that around 1/3 of the size of my entire DB was just logged webhook requests!

Would be great if there were a config setting for number of days' worth of requests to log, with older ones being flushed out at GC or something.

**Comment by brandonkelly** [comment, marked as answer]
We just released Webhooks 2.4.0, which now purges old webhook request logs during [garbage collection](https://craftcms.com/docs/3.x/gc.html).

By default any requests over 7 days old will be deleted, and you can configure that duration (or disable the purging entirely) via the new **Purge Duration** plugin setting.

**Comment by mildlygeeky** [comment]
Thank you!!!

Patrick Harrington
617.505.4456
________________________________
